### PR TITLE
✨ Add a more realistic sample application to /examples

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

We wanted a sample that was simpler to modify and was more realistic to real world scenarios, including Flutter v2 routing, error handling, and http tracking.

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests 
- [ ] This pull request references a Github or JIRA issue

### CI Configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ This folder contains examples of using the Datadog Flutter Plugin in more compli
 
 This includes:
 
+* [Simple Example](./simple_example) - Includes eamples of GoRouter integration, distributed tracing, interaciton tracking, and error / crash reporting.
```
Comment (brett0000FF) [inline examples/README.md:7]:
```suggestion
* [Simple Example](./simple_example) - Includes examples of GoRouter integration, distributed tracing, interaction tracking, and error/crash reporting.
```

File: examples/simple_example/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Datadog Flutter Plugin Simple Example
```
Comment (brett0000FF) [inline examples/simple_example/README.md:1]:
```suggestion
# Datadog Flutter Plugin -  Simple Example
```

File: examples/simple_example/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+This is a slighly more realistic / complicated example of using the Datadog Flutter Plugin
+in a real applicaiton scenario.  This includes
```
Comment (brett0000FF) [inline examples/simple_example/README.md:4]:
```suggestion
This is a more realistic example of how you use the Datadog Flutter Plugin in a real-world scenario.  This includes:
```

File: examples/simple_example/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+    
+    * Demonstrating using GoRouter with the DatadogNavigationObserver
+    * Automatic network tracing with `datadog_tracking_http_client`
+    * User interaction tracking with RumUserActionDetector
+    * Error / Crash handling with manually reported errors
```
Comment (brett0000FF) [inline examples/simple_example/README.md:9]:
```suggestion
    * Using GoRouter with the DatadogNavigationObserver
    * Automatic network tracing with `datadog_tracking_http_client`
    * User interaction tracking with RumUserActionDetector
    * Error/Crash handling with manually reported errors
```

File: examples/simple_example/ios/Runner/Assets.xcassets/LaunchImage.imageset/README.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Launch Screen Assets
```
Comment (brett0000FF) [inline examples/simple_example/ios/Runner/Assets.xcassets/LaunchImage.imageset/README.md:1]:
```suggestion
# Customizing Launch Screen Assets
```

File: examples/simple_example/ios/Runner/Assets.xcassets/LaunchImage.imageset/README.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+
+You can customize the launch screen with your own desired assets by replacing the image files in this directory.
+
+You can also do it by opening your Flutter project's Xcode project with `open ios/Runner.xcworkspace`, selecting `Runner/Assets.xcassets` in the Project Navigator and dropping in the desired images.
```
Comment (brett0000FF) [inline examples/simple_example/ios/Runner/Assets.xcassets/LaunchImage.imageset/README.md:5]:
```suggestion
You can also customize the launch screen directly in your Flutter project's Xcode project with the following steps:

1. Open your Xcode project:`open ios/Runner.xcworkspace`. 
2. Select `Runner/Assets.xcassets` in the Project Navigator.
3.  Place your images into the `Runner/Assets.xcassets` location.
```
Just a suggestion to break this into numbered steps.

**Review by brett0000FF — APPROVED** [review]
Looks good. Just added a couple suggestions.

I'd also recommend changing section headers to sentence case, as that is our standard for docs. I didn't make those suggestions in this PR though because there are sections headers in unchanged parts of the files.

File: examples/simple_example/android/local.properties
Comment (0xnm) [inline examples/simple_example/android/local.properties:1]: this file shouldn't be commited, it is local

**Review by 0xnm — APPROVED** [review]
lgtm! I've added a comment about `local.properties` file which shouldn't be commited.

**Pull request merged**